src/heuristic/HeuristicMapper.cpp

```diff
@@ -7,29 +7,36 @@
 
 #include "utils.hpp"
 
+#include <algorithm>
 #include <chrono>
 
 void HeuristicMapper::map(const Configuration& configuration) {
   if (configuration.dataLoggingEnabled()) {
     dataLogger = std::make_unique<DataLogger>(configuration.dataLoggingPath,
                                               *architecture, qc);
   }
-  results        = MappingResults{};
-  results.config = configuration;
-  auto& config   = results.config;
+
+  tightHeur         = isTight(configuration.heuristic);
+  fidelityAwareHeur = isFidelityAware(configuration.heuristic);
+
+  results            = MappingResults{};
+  results.config     = configuration;
+  const auto& config = results.config;
   if (config.layering == Layering::OddGates ||
       config.layering == Layering::QubitTriangle) {
     throw QMAPException("Layering strategy " + toString(config.layering) +
                         " not suitable for heuristic mapper!");
   }
-  if (config.considerFidelity && !architecture->isFidelityAvailable()) {
+  if (fidelityAwareHeur && !architecture->isFidelityAvailable()) {
     throw QMAPException("No calibration data available for this architecture!");
   }
-  if (config.considerFidelity && config.lookahead) {
-    throw QMAPException("Lookahead is not yet supported for heuristic mapper "
-                        "using fidelity-aware mapping!");
+  if (fidelityAwareHeur && !isFidelityAware(config.lookaheadHeuristic) &&
+      config.lookaheadHeuristic != LookaheadHeuristic::None) {
+    throw QMAPException("Fidelity-aware heuristics may only be used with "
+                        "fidelity-aware lookahead heuristics (or no "
+                        "lookahead)!");
   }
-  if (config.considerFidelity && config.teleportationQubits > 0) {
+  if (fidelityAwareHeur && config.teleportationQubits > 0) {
```

Comment:
At some point, we should probably refactor all these checks into a separate function like `checkParameters` (or similar) so that these additional conditions do not further bloat the `map` function.

Reply:
Great idea, I just implemented that and pushed it to this PR.